### DISCUSSION TITLE
Using B-L475E-IOT01A with arduino-cli

### DISCUSSION THREAD

**Discussion by SebastianZug** [stm32duino/Arduino_Core_STM32#2255] [General]
Hi,

I want to use Arduino command line interface (arduino-cli) for compiling code and flashing an B-L475E-IOT01A. Consequently, I installed the `STMicroelectronics:stm32` packages.

```
> ./arduino-cli board listall stm
Discovery                    STMicroelectronics:stm32:Disco
Elecgator boards             STMicroelectronics:stm32:Elecgator
Electronic speed controllers STMicroelectronics:stm32:ESC_board
Eval                         STMicroelectronics:stm32:Eval
Garatronic-McHobby           STMicroelectronics:stm32:Garatronic
Generic Flight Controllers   STMicroelectronics:stm32:GenFlight
```

But the actual compilation does not work. I assume that my reference on the specific board is wrong when starting the compile process:

```
./arduino-cli compile --fqbn STMicroelectronics:stm32:Disco:B-L475E-IOT01A ArduinoExample
Error during build: Invalid FQBN: invalid config option: B-L475E-IOT01A
```

In contrast, inside the Arduino IDE you have to select 
 `Board : Discovery` and `Board part number: B-L475E-IOT01A`.  But I have no idea how to transfer these two information while calling compilation.
 
 Many thanks in advance!
 
Sebastian

**Comment by fpistm** [comment]
You can enable some verbose in the ide then you will see the fqbn.

**Reply by SebastianZug** [reply]
By parsing the outputs of the Arduino IDE, I found this solution:

```
./arduino-cli compile --fqbn=STMicroelectronics:stm32:Disco:pnum=B_L475E_IOT01A ArduinoExample 
```

You saved my day. Many thanks!

**Reply by fpistm** [reply]
Welcome
